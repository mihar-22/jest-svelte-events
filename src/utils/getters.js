export const getListener = component => {
  const listeners = global.SVELTE_LISTENERS
  const listener = listeners[component]
  if (!listener) {
    throw Error(
      'This component has not been registered and has no listeners, ' +
    'setup the listener before calling the matcher.\n\n' +
    'Eg: listen(component, \'myEvent\')\n\n'
    )
  }
  return listener
}

export const getEventData = (component, event) => {
  const listener = getListener(component)
  const eventData = listener[event]
  if (!eventData) {
    throw Error(
    `There are no listeners for the event \`${event}\` on this component, ` +
    'setup the listener before calling the matcher.\n\n' +
    'Eg: listen(component, \'myEvent\')\n\n'
    )
  }
  return eventData
}
