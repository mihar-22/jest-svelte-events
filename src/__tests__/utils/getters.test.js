import '../../'
import { getListener, getEventData } from '../../utils/getters'

import { render } from '@testing-library/svelte'

import Comp from '../fixtures/Comp.svelte'

describe('utils', () => {
  describe('getters', () => {
    describe('getListener', () => {
      test('returns listener', () => {
        const { component } = render(Comp)
        listen(component, 'e1')
        expect(getListener(component)).toMatchSnapshot()
      })

      test('throws error when component has not been registered', () => {
        const { component } = render(Comp)
        expect(() => {
          expect(component).toHaveFiredEvent('e1')
        }).toThrowErrorMatchingSnapshot()
      })
    })

    describe('getEventData', () => {
      test('returns event data', () => {
        const { component } = render(Comp)
        listen(component, 'e1')
        expect(getEventData(component, 'e1')).toMatchSnapshot()
      })

      test('throws error when event has no listener registered', () => {
        const { component } = render(Comp)
        listen(component, 'e1')
        expect(() => {
          expect(component).toHaveFiredEvent('e2')
        }).toThrowErrorMatchingSnapshot()
      })
    })
  })
})
