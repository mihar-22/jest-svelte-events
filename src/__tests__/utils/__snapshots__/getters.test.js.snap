// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils getters getEventData returns event data 1`] = `
Object {
  "calls": 0,
  "payloads": Array [],
}
`;

exports[`utils getters getEventData throws error when event has no listener registered 1`] = `
"There are no listeners for the event \`e2\` on this component, setup the listener before calling the matcher.

Eg: listen(component, 'myEvent')

"
`;

exports[`utils getters getListener returns listener 1`] = `
Object {
  "e1": Object {
    "calls": 0,
    "payloads": Array [],
  },
  "stack": Array [],
}
`;

exports[`utils getters getListener throws error when component has not been registered 1`] = `
"This component has not been registered and has no listeners, setup the listener before calling the matcher.

Eg: listen(component, 'myEvent')

"
`;
